From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frank Chen <frankc@nvidia.com>
Date: Tue, 29 Nov 2022 11:17:21 -0800
Subject: [PATCH 015/241] NVIDIA: SAUCE: v4l2-core: Add new packed v4l2 pixel
 format

Add pacekd 10-bit formats to v4l2 pixel formats list.
X(Bayer)10 packs 3 pixels to 4 bytes with two MSB bits
set to zero. All four corresponding phases are added.

Bug 200602400

Change-Id: I5f1f9d98699b5e2b40c909184c6505593ad96380
Signed-off-by: Frank Chen <frankc@nvidia.com>
(cherry picked from a4188c9565a9b0961fa7fb3f1b74911cd0b18858)
Reviewed-by: Laxman Dewangan <ldewangan@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/linux-stable/+/2897715
Tested-by: Abhilash G <abhilashg@nvidia.com>
Reviewed-by: Abhilash G <abhilashg@nvidia.com>

diff --git a/drivers/media/v4l2-core/v4l2-ioctl.c b/drivers/media/v4l2-core/v4l2-ioctl.c
index 5e019c10afa34..ff16bc5a3d49d 100644
--- a/drivers/media/v4l2-core/v4l2-ioctl.c
+++ b/drivers/media/v4l2-core/v4l2-ioctl.c
@@ -1403,6 +1403,18 @@ static void v4l_fill_fmtdesc(struct v4l2_fmtdesc *fmt)
 	case V4L2_PIX_FMT_IPU3_SGBRG10: descr = "10-bit bayer GBRG IPU3 Packed"; break;
 	case V4L2_PIX_FMT_IPU3_SGRBG10: descr = "10-bit bayer GRBG IPU3 Packed"; break;
 	case V4L2_PIX_FMT_IPU3_SRGGB10: descr = "10-bit bayer RGGB IPU3 Packed"; break;
+	case V4L2_PIX_FMT_XBGGR10P:
+		descr = "10-bit Bayer BGGR(10-10-10-2)";
+		break;
+	case V4L2_PIX_FMT_XGBRG10P:
+		descr = "10-bit Bayer GBRG(10-10-10-2)";
+		break;
+	case V4L2_PIX_FMT_XGRBG10P:
+		descr = "10-bit Bayer GRBG(10-10-10-2)";
+		break;
+	case V4L2_PIX_FMT_XRGGB10P:
+		descr = "10-bit Bayer RGGB(10-10-10-2)";
+		break;
 	case V4L2_PIX_FMT_SBGGR10ALAW8:	descr = "8-bit Bayer BGBG/GRGR (A-law)"; break;
 	case V4L2_PIX_FMT_SGBRG10ALAW8:	descr = "8-bit Bayer GBGB/RGRG (A-law)"; break;
 	case V4L2_PIX_FMT_SGRBG10ALAW8:	descr = "8-bit Bayer GRGR/BGBG (A-law)"; break;
-- 
2.43.0

